sdk/core/azure-core/tests/test_pyodide_transport.py

```diff
@@ -0,0 +1,194 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+"""Tests that mock the browser layer."""
+import asyncio
+import sys
+from typing import NamedTuple
+from unittest import mock
+
+import pytest
+from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline._base_async import AsyncPipeline
+from azure.core.pipeline.policies._retry_async import AsyncRetryPolicy
+from azure.core.rest import HttpRequest
+
+PLACEHOLDER_ENDPOINT = "https://my-resource-group.cognitiveservices.azure.com/"
+
+
+class TestPyodideTransportClass:
+    """Unittest for the Pyodide transport."""
+
+    @pytest.fixture()
```

Comment:
are you able to use the core testserver for pyodide? My guess is not bc it's a flask app, but would be great if this could be added to the testserver

Reply:
I wish, but I can't call `pyfetch` without mocking it.